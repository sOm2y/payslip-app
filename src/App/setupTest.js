//@flow

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// enable using of expect(wrapper).toContainReact(<SomeComponent/>)
import 'jest-enzyme';

configure({ adapter: new Adapter() });
