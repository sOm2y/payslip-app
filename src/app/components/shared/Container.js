//@flow

import React from 'react';
import type { Node } from 'react';

import Paper from 'material-ui/Paper';
import { withStyles } from 'material-ui/styles';

const styles = {
  container: {
    padding: 32,
    maxWidth: 360
  }
};

class Container extends React.PureComponent<{ children: Node, classes: { container: string } }> {
  render() {
    const { classes, children, ...others } = this.props;

    return (
      <Paper {...others} className={classes.container}>
        {children}
      </Paper>
    );
  }
}

export default withStyles(styles)(Container);
