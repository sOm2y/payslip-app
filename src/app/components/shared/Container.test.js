//@flow

import React from 'react';
import Container from './Container';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import '../../setupTest'; // enzyme setup

describe('Container', () => {
  it('should renders without crashing even for shallow renderer', () => {
    shallow(<Container />);
  });

  it('should render children', () => {
    const Children = () => <div>foo</div>;

    const result = shallow(
      <Container>
        <Children />
      </Container>
    );

    expect(result).toContainReact(<Children />);
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<Container>hello</Container>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
