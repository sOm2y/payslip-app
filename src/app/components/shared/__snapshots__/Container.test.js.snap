// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match snapshot 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-shadow2-6 MuiPaper-rounded-3 Container-container-1"
>
  hello
</div>
`;
